lib/shared.bash

```diff
@@ -322,6 +322,31 @@ function retry {
   done
 }
 
+# Expands the env vars in a string, using envsubst if present, falling back to
+# eval when it's missing.
+function expand_var() {
+  # Try to use the safest approach possible
+  if command -v envsubst > /dev/null; then
```

Comment:
It's occurred to me just now that some folk could potentially be leaning on that other pre-existing `eval` method in ways that had not occurred to me before

eg.

```yaml
    plugins:
      - docker-compose: 
          expand-volume-vars: true
          build: target
          run: target
          volumes: 
            - "$(command -v oh-no):/bin/oh-no"
```

Assuming that's possible, I guess we'd want to tie this to an option rather than just finding it on the system. Not sure if it would make sense to have an option for each case, or just a single option that covers all cases.

What do you think?

Reply:
that is indeed a very good point... a good middle-ground could be changing the existing options to not be booleans but instead one of `true/false/eval/envsubst` so that users can decide what to do on each case. A modular implementation of that would be to change the `expand_var` function to accept the value of the corresponding setting and use the corresponding implementation. What do you think?

Reply:
Sounds pretty sensible 👍  I'll find some time this week to update again

Reply:
Hey @lparry! Thanks for your contributions here! Are you able to update this PR based on the previous recommendations? 🙏  thanks!